compiler/rustc_hir_analysis/src/check/intrinsic.rs

```diff
@@ -54,23 +60,22 @@ fn equate_intrinsic_type<'tcx>(
         && gen_count_ok(own_counts.types, n_tps, "type")
         && gen_count_ok(own_counts.consts, n_cts, "const")
     {
-        let it_def_id = it.owner_id.def_id;
         let _ = check_function_signature(
             tcx,
-            ObligationCause::new(it.span, it_def_id, ObligationCauseCode::IntrinsicType),
-            it_def_id.into(),
+            ObligationCause::new(span, def_id, ObligationCauseCode::IntrinsicType),
+            def_id.into(),
             sig,
         );
     }
 }
 
 /// Returns the unsafety of the given intrinsic.
-pub fn intrinsic_operation_unsafety(tcx: TyCtxt<'_>, intrinsic_id: DefId) -> hir::Unsafety {
+pub fn intrinsic_operation_unsafety(tcx: TyCtxt<'_>, intrinsic_id: LocalDefId) -> hir::Unsafety {
```

Comment:
The attribute only exists for normal intrinsics, because we have no `unsafe` marker in extern block's function items. I'll make sure it can't be applied to intrinsics with fallback bodies.

Reply:
Okay, that makes sense. Can you add a test that an unsafe intrinsic with a fallback body is actually considered unsafe?

Reply:
This is also probably worth documenting somewhere. I don't want someone to copy an old intrinsic adding a fallback body and accidentally make it safe...

Reply:
I documented it, and you can't accidentally screw up, the compiler checks it against the internal set here. Before [585e76a](https://github.com/rust-lang/rust/pull/120500/commits/585e76a78cca40677c5045189a0d607c82e514cd) all `rustc_intrinsic`s were unsafe by mistake.

Reply:
Perfect then